test/integration_test/lib/util/test_union_find.py

```diff
@@ -0,0 +1,18 @@
+from lib.util.union_find import *
+import pytest
+
+def test_union_find(self):
```

Comment:
The class name should be `TestUnionFind` and have one test method for each method that is getting tested.